### ISSUE TITLE
TypeScript reports incorrect error

### ISSUE THREAD

**Issue by johndeighan** [microsoft/TypeScript#61563]
### 🔎 Search Terms

type narrowing

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about type narrowing

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.9.0-dev.20250410#code/EQVwzgpgBGAuBOBLAxrYBuAUMg9gOzigBsARAQ1jIC4oAKPEAWwCMJ4oAfGBRPAc05QQeACYQAZrwgiAlAG0AulAC8UOZgCQAFgBMAGk0ByMs2SGDG4WMl5pmhZmz5CRAEo4A7jThJ+ilcTklAB0jGQADrS0iLAQjDIqAHxQAN6aiOJ0MXEqyqpWElKyUPAQsCDweFCGhlgaGXSwAJ7hEDiZ2Yy5qoYMLGyGCaXllVCdwbA4AGKIAB7StDoydQ20za3tY7FdeT0+vHyDJWUVVZ1YAL4yjrgEOEQQwSKI8LRuntdAA

### 💻 Code

```ts
"use strict";
const lData: (number | string | undefined)[] = [
	42,
	'abc',
	undefined
	]

const lRow: string[] = lData.map((item) => {
	if (item === undefined) return '';
	if (typeof item === 'number') return item.toFixed(2);
	if (typeof item === 'string') return item;
})

console.dir(lRow)```


### 🙁 Actual behavior

The above code results in errors, even though the mapping function being used can never return an undefined value. Oddly, if I use a switch statement instead of a series of if statements, no errors are reported:

```
"use strict";
const lData: (number | string | undefined)[] = [
	42,
	'abc',
	undefined
	]

const lRow: string[] = lData.map((item) => {
	switch(typeof item) {
		case 'undefined': {
			return '';
		}
		case 'number': {
			return item.toFixed(2);
		}
		case 'string': {
			return item;
		}
	}
})

console.dir(lRow)

```

### 🙂 Expected behavior

There should be no errors in either case since the items being mapped over can only be undefined, a string or a number. Note that in both of the examples, there is no "else" case.

### Additional information about the issue

_No response_

**Comment by nmain** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/21985.  My search terms were `if never in:title`.  It works if you explicitly annotate the `never`:

```typescript
const lRow: string[] = lData.map((item) => {
	if (item === undefined) return '';
	if (typeof item === 'number') return item.toFixed(2);
	if (typeof item === 'string') return item;
	item satisfies never;
	return item;
})
```

**Comment by johndeighan** [comment]
Nice to have a hacky work-around, though I think it would be better to actually fix the problem. Meanwhile, I'll just use a switch statement since it doesn't have this problem.
